pkg/client/results/processing.go

```diff
@@ -130,6 +132,49 @@ func (i *Item) GetSubTreeByName(root string) *Item {
 	return nil
 }
 
+// manualResultsAggregation is custom logic just for aggregating results for the top level summary
+// when the plugin is providing the YAML results manually. This is required in (at least) some cases
+// such as daemonsets when each plugin-node will have a result that needs bubbled up to a single,
+// summary Item. This is so that in large clusters you dont have a single plugin have results that
+// scale linearly with the number of nodes and may become unreasonable to show the user.
+//
+// If there is only one top level item, its status is returned. Otherwise a human readable string
+// is produced to show the counts of various values. E.g. "3 passed/2 failed/1 custom msg". Avoiding
```

Comment:
@zubron What do you think about this solution?

If we do something like this then we can summarize any custom status the user puts on the items. It seems like the only other alternative is to force them into a very narrow set of allowed values (passed/failed/skipped) which seemed to me to miss the point of allowing completely custom results from the plugin.

Reply:
I think this is a reasonable approach. You make a good point about the narrow set of allowed values, I think it will be preferable to allow custom statuses there :+1: